ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java

```diff
@@ -213,7 +213,7 @@ private void validateConfig(final GraphQLConfiguration config) {
   }
 
   private Optional<String> getAndValidateHostHeader(final RoutingContext event) {
-    final Iterable<String> splitHostHeader = Splitter.on(':').split(event.request().host());
+    final Iterable<String> splitHostHeader = Splitter.on(':').split(event.request().getHeader(HttpHeaders.HOST));
```

Comment:
This is an optional header, I think we need some null checking to do this safely, otherwise we have test regressions.